### ISSUE TITLE
Can't Watch Variables during Debug

### ISSUE THREAD

**Issue by areller** [OmniSharp/omnisharp-vscode#2912]
## Issue Description ##
I try to debug the extension with VSCode, but whenever I enter debug mode, I can't view the values of the variables when I hover over them.

<img width="869" alt="image" src="https://user-images.githubusercontent.com/15989650/54475789-4d8f1a00-47fe-11e9-9408-2c3310f57327.png">

(I hover over `const extensionId` but nothing happens)

## Steps to Reproduce ##

1. Clone the repository
2. Open it with VSCode
3. Run `npm install`
4. Put a breakpoint in `main.ts`.`active`.`const extensionId = ...`
5. Press F5 to start the debugger

## Expected Behavior ##

I want to be able to watch/view variables during debug

<img width="544" alt="image" src="https://user-images.githubusercontent.com/15989650/54475826-ce4e1600-47fe-11e9-8f83-55309e26847c.png">

This is for a different project.
You can see that when I hover over `num`, it displays its value.

## Actual Behavior ##

Hovering over variables does nothing.

### OmniSharp log ###
<details>Post the output from Output-->OmniSharp log here</details>

### C# log ###
<details>Post the output from Output-->C# here</details>

## Environment information ##

**VSCode version**: 1.32.3
**C# Extension**: 1.17.1

<details><summary>Mono Information</summary>
        OmniSharp using built-in mono</details>
<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview3-010431
 Commit:    d72abce213

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.0.100-preview3-010431/

Host (useful for support):
  Version: 3.0.0-preview3-27503-5
  Commit:  3844df9537

.NET Core SDKs installed:
  2.2.103 [/usr/local/share/dotnet/sdk]
  3.0.100-preview3-010431 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview3-19153-02 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview3-27503-5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|code-spell-checker|streetsidesoftware|1.6.10|
|cpptools|ms-vscode|0.21.0|
|csharp|ms-vscode|1.17.1|
|csharpextensions|jchannon|1.3.0|
|gitlens|eamodio|9.5.1|
|Go|ms-vscode|0.9.2|
|Lua|keyring|0.0.9|
|python|ms-python|2019.2.5558|
|vscode-docker|PeterJausovec|0.6.0|
|vscode-lua|trixnz|0.12.4|
|vscode-proto3|zxh404|0.2.2|
|vscodeilviewer|josephwoodward|0.0.1|;

</details>

**Comment by gregg-miskelly** [comment]
@areller you opened this issue here because you are having trouble debugging the C# extension, is that correct? This probably isn't the right place for this issue. Are you opening this repo as a folder and hitting F5?

**Comment by areller** [comment]
Yes. I'm having trouble debugging the C# extension.
I open the omnisharp-vscode folder, via VSCode, after I've cloned this project. Then, I do `npm install` and then I press F5 to launch the debugger, which leads me to the issue itself.
I understand what you're saying. It might not be related directly to the omnisharp-vscode extension, but rather to the component that is responsible for TypeScript debugging, but I want to make sure.

**Comment by akshita31** [comment]
@areller I have also experienced a similar issue recently and I think this is happening since the time we enabled webpack for the extension.

For now can you try the following workaround: Replace the "/dist/extension" in [package.json](https://github.com/OmniSharp/omnisharp-vscode/blob/master/package.json#L30) with "out/src/main" and see if it helps ?

cc @rchande

**Comment by areller** [comment]
@akshita31 Changing the `main` field to `out/src/main` **together with** changing `${workspaceRoot}/dist/*.js` to `${workspaceRoot}/out/*.js` in `.vscode/launch.json` fixed it.  
Debugging works now.

Why do you think it happens with webpack? Is it because it doesn't generate source maps properly?
And out of curiosity, why use webpack instead of just creating a vscode build task that runs `tsc`?

Do you know whether there is an existing issue/effort for fixing this problem?

**Comment by akshita31** [comment]
@areller If you look at the issues in the repo a number of them say, "Extension causes high cpu load". On analysing the traces we realised that the code loading of the extension was taking a lot of time. So we bundled the extension using webpack that performs tree shaking and has significantly reduces the code loading times.

As to why it happens with webpack, I am also guessing that there is a problem with source maps but I am not sure how to resolve/debug it.

We dont have any issue tracking the problem as of now. We can use this one to keep track of it.
